tests/python_package_test/test_engine.py

```diff
@@ -828,3 +829,20 @@ def preprocess_data(dtrain, dtest, params):
         results = lgb.cv(params, dataset, num_boost_round=10, fpreproc=preprocess_data)
         self.assertIn('multi_logloss-mean', results)
         self.assertEqual(len(results['multi_logloss-mean']), 10)
+
+    @unittest.skipIf(psutil.virtual_memory().free / 1024 / 1024 / 1024 < 3, 'not enough RAM')
```

Comment:
To be honest, I don't know whether it's cross-platform or not.
On my Windows with Python 3.6 it shows approximately 2 GB.


Or maybe it's better to wrap it in
```
try:
   …
except MemoryError:
   pass
```

Reply:
It seems some tests throw memory error..

Reply:
Yeah!..
I'll add `try/except`, but I don't know should I bump free memory requirement too (and to what value) or not. What do you think?

Reply:
@StrikerRUS I think we can keep them both, can use a memory size larger than 4GB and smaller than 8GB, such as 7GB?

Reply:
@guolinke OK. I agree with this. One more question: did you mean 7GB of **free** RAM or overall?

Reply:
https://github.com/Microsoft/LightGBM/blob/cbae0c1b1681dedd6a519759c13964f4d3ee7e35/tests/python_package_test/test_engine.py#L833
Unfortunately, it caused even more MemoryErrors. Will try with 4 GB of free volume.

Reply:
There is no difference in results for 4, 5, 6 GB. Only requirement of 7 GB of free memory makes this test to be skipped which results in green CI checks.
https://github.com/Microsoft/LightGBM/blob/e72ee0335cb5c05462ce3dddbe517d1eb4447bc8/tests/python_package_test/test_engine.py#L833
Seems that this `skipIf` is not useful actually...